### PR TITLE
DEVPROD-8256: Implement GitHub Permission Groups tab

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-8256

### Description
Implements the GitHub Permission Groups tab.

### Screenshots
Screenshot 2024-07-10 at 9 51 01 AM

### Testing
- Cypress tests
- Jest tests for transformers

### CODE REVIEW

File: apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/FieldTemplates.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+> = ({ formData, properties }) => {
+  const [permissionType, permissionValue] = getFields(
+    properties,
+    formData.isDisabled,
+  );
+  return (
+    <RowContainer>
+      {permissionType}
+      {permissionValue}
+    </RowContainer>
```
Comment (sophstad) [inline apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/FieldTemplates.tsx:25]: Should we validate against duplicate keys or key/value pairs in the table?
image

File: apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/FieldTemplates.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import { Body } from "@leafygreen-ui/typography";
+import { ArrayFieldTemplateProps, ObjectFieldTemplateProps } from "@rjsf/core";
+import { PlusButton } from "components/Buttons";
+import Icon from "components/Icon";
+import { getFields } from "components/SpruceForm/utils";
+import { BaseTable } from "components/Table/BaseTable";
+import { size, tableColumnOffset } from "constants/tokens";
+import { Unpacked } from "types/utils";
+
+export const PermissionObjectFieldTemplate: React.FC<
```
Comment (sophstad) [inline apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/FieldTemplates.tsx:14]: Nice, would be cool to genericize this at some point so we can move away from having inputs with repetitive labels!

File: apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/getFormSchema.tsx [hunk 3]
```diff
@@ -1,15 +1,145 @@
...
+  uiSchema: {
+    permissionGroups: {
+      "ui:orderable": false,
+      "ui:addButtonText": "Add permission group",
+      "ui:descriptionNode": (
+        <>
+          <StyledDescription>
+            Create permission groups containing a set of permissions for
+            generated tokens. Permission groups can be applied to one or more
+            requester types in the GitHub App Settings Tab. When assigned to a
```
Comment (sophstad) [inline apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/getFormSchema.tsx:86]: [nit]
```suggestion
            requester types in the GitHub App Settings tab. When assigned to a
```

File: apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/FieldTemplates.tsx (hunk 1 above)
Comment (minnakt) [reply apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/FieldTemplates.tsx:25]: added validation but would note that the error message messes up the alignment with the trash can icon, but it's not really fixable ☹️

File: apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/getFormSchema.tsx [hunk 4]
```diff
@@ -1,15 +1,146 @@
...
+  uiSchema: {
+    permissionGroups: {
+      "ui:orderable": false,
+      "ui:addButtonText": "Add permission group",
+      "ui:descriptionNode": (
+        <>
+          <StyledDescription>
+            Create permission groups containing a set of permissions for
+            generated tokens. Permission groups can be applied to one or more
+            requester types in the GitHub App Settings tab. When assigned to a
```
Comment (sophstad) [inline apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/getFormSchema.tsx:87]: Could add a link to the app settings tab if you want!

**Review by sophstad — APPROVED** [review]

**Pull request merged**